plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/tasks/BaseKtLintCheckTask.kt

```diff
@@ -119,15 +120,29 @@ abstract class BaseKtLintCheckTask @Inject constructor(
         .from({ sourceFiles.asFileTree.matching(patternFilterable) })
 
     /**
-     * Sets the source from this task.
+     * Sets the source for this task from a given file tree.
+     *
+     * Filters from configuration will be applied
      *
      * @param source given source objects will be evaluated as per [org.gradle.api.Project.file].
      */
-    fun setSource(source: Any): BaseKtLintCheckTask {
-        sourceFiles = objectFactory.fileCollection().from(source)
+    fun setSource(source: FileTree): BaseKtLintCheckTask {
+        sourceFiles = objectFactory
+            .fileCollection()
+            .from({ source.matching(patternFilterable) })
```

Comment:
does this mean the filter is applied twice? since the provider on line 120 also applies the same filter?

Reply:
Actually this is my first contact with gradle internals so I don't fully understand when these methods are called.
From diffing the debug output of Windows and Linux runs I got the impression, that filesystem changes (inotify et al triggered while generating the stubs) are somehow handled differently.